client/asset/eth/multirpc.go

```diff
@@ -1096,18 +1100,12 @@ func (m *multiRPCClient) withAll(
 	if atLeastOne {
 		return nil
 	}
-	if errs == nil {
-		return errors.New("all providers in a failed state")
-	}
```

Comment:
RE: https://github.com/decred/dcrdex/pull/2803/files#r1641230821

This error check is also buggy, hence the refactor here. Plus the related TODO.

Reply:
The "all providers in a failed state" error didn't indicate that all requests failed. It indicated that no requests were made, since all providers were in a failed state and could not be used. But I do see how that could have been wrong if e.g. we had discarded an error. Regardless, if we get this far with `errs == nil`, that still indicates an error, I think.